store/store_action.go

```diff
@@ -319,6 +320,15 @@ type SnapResourceResult struct {
 	CreatedAt    string
 }
 
+// ResourceToComponentType returns a validated component type from a resource type.
+func ResourceToComponentType(resType string) (snap.ComponentType, error) {
```

Comment:
This could replace some code in snapstate's `componentSetupFromResource` function.